src/actions/read/auction.ts

```diff
@@ -1,18 +1,67 @@
-import { ContractResult, IOState, PstAction } from '../../types';
-import { calculateMinimumAuctionBid } from '../../utilities';
+import {
+  AuctionSettings,
+  ContractResult,
+  IOState,
+  PstAction,
+} from '../../types';
+import {
+  calculateMinimumAuctionBid,
+  generateAuctionPermutations,
+  getInvalidAjvMessage,
+} from '../../utilities';
+import { validateGetAuction } from '../../validations.mjs';
 
 declare const SmartWeave: any;
 declare const ContractError;
 
+export class GetAuction {
+  function = 'getAuction';
+  name: string;
+
+  constructor(input: any) {
+    // validate using ajv validator
+    if (!validateGetAuction(input)) {
+      throw new ContractError(getInvalidAjvMessage(validateGetAuction, input));
+    }
+    const { name } = input;
+    this.name = name.trim().toLowerCase();
+  }
+}
+
 export const getAuction = (
   state: IOState,
-  { input: { name } }: PstAction,
+  { caller, input }: PstAction,
 ): ContractResult => {
-  const { auctions, settings } = state;
+  const { name } = new GetAuction(input);
+  const { auctions, settings, fees } = state;
   const auction = auctions[name.toLowerCase().trim()];
 
   if (!auction) {
-    throw new ContractError(`No live auction exists for ${auction}`);
+    // No auction? generate the relevant auctions permutations for the name (lease lengths + permabuy)
```

Comment:
why?

Reply:
Pricing api for getting the price when registering

Reply:
hm - trying to decide if we should put this under `getAuctionPrices` API  and have `getAuction` keep throwing the error if it's not already an auction

Reply:
in a RESTFUL sense

`v1/contract/:id/auctions/:name` -> currently returns 404 when no auction exists (e.g. the contract throws)

what we have here would change that to something like

`/v1/contract/:id/auctions/:name` -> returns list of prices for the name _should_ it be auctioned (if it's not already) 🤔

Reply:
IMMHO the simple interface/complex implementation is the way to go here.

Implementing the interface like this is attractive to me on the front end, because i can use the same api to get both pricing for an auction that exists and does not exist (expected costs when performing a transaction).

As for implementing it on the service, again i think its attractive for keeping all the pricing logic in the contract do exclude other components needed to worry about the dependencies there.

The consumer of this data needs to implement error handling of the contract throwing regardless - they can choose to use the data that it returns or not, but I figure its better to give the option, especially since its usefull data.

And I dont think the objects will be confused - they are constructed completely differently.